thavalon-server/src/lobby/lobby_impl.rs

```diff
@@ -170,14 +210,79 @@ impl Lobby {
         LobbyResponse::Standard(Ok(()))
     }
 
-    /// [WIP] - Starts a game.
+    /// Handles a change to the player list, due to a player joining or leaving the game.
+    async fn on_player_list_change(&mut self) {
+        // Only broadcast to players if the game hasn't started yet.
+        // TODO: Maybe a helpful message to players that someone has disconnected.
+        // Would need to have a way to lookup name by the disconnected client ID.
+        if self.status == LobbyState::Lobby {
+            let current_players = self.builder.as_ref().unwrap().get_player_list();
+            self.broadcast_message(&OutgoingMessage::PlayerList(current_players.to_vec()))
+                .await;
+        }
+    }
+
+    // Handles dealing with a disconnected player.
+    // If the lobby isn't in progress or done, a disconnect should remove the player.
+    // Otherwise, nothing happens.
+    async fn on_player_disconnect(&mut self, client_id: String) -> LobbyResponse {
+        log::info!(
+            "Client {} has disconnected from game {}.",
+            client_id,
+            self.friend_code
+        );
+
+        // If we're in the lobby phase, a disconnect counts as leaving the game.
+        if self.status == LobbyState::Lobby {
+            self.remove_player(client_id).await;
+        }
+
+        LobbyResponse::Standard(Ok(()))
+    }
+
+    /// Starts the game and updates statuses
     async fn start_game(&mut self) -> LobbyResponse {
-        // TODO: This function is a stub that should be expanded.
-        // It is currently here to remove compiler warnings.
-        let _ = self.database_game.start_game().await;
+        // The only thing that can fail is updating the database. In this case,
+        // the lobby is probably dead, so panic to blow up everything.
+        if let Err(e) = self.database_game.start_game().await {
+            log::error!("Error while starting game {}. {}", self.friend_code, e);
+            panic!();
```

Comment:
will this take down server completely?

Reply:
No, it'll only take down the lobby. It's also changed in the new version I think.